src/raftstore/store/store.rs

```diff
@@ -944,6 +927,35 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         Ok(())
     }
 
+    fn validate_region(&self, msg: &RaftCmdRequest) -> Result<()> {
+        let region_id = msg.get_header().get_region_id();
+        let peer_id = msg.get_header().get_peer().get_id();
+
+        let peer = match self.region_peers.get(&region_id) {
+            Some(peer) => peer,
+            None => return Err(Error::RegionNotFound(region_id)),
+        };
+        if !peer.is_leader() {
+            return Err(Error::NotLeader(region_id, peer.get_peer_from_cache(peer.leader_id())));
+        }
+        if peer.peer_id() != peer_id {
+            return Err(box_err!("mismatch peer id {} != {}", peer.peer_id(), peer_id));
+        }
+
+        let res = peer.check_epoch(msg);
+        if let Err(Error::StaleEpoch(msg, mut new_regions)) = res {
+            // Attach next region, which maybe splitted from current region.
+            if let Some((_, &next_region_id)) = self.region_ranges
+                .range(Excluded(&enc_end_key(peer.region())), Unbounded::<&Key>)
+                .next() {
+                let next_region = self.region_peers[&next_region_id].region();
```

Comment:
It won't matter even if next region is not actually splitted from current region. Whenever TiDB receives a region meta newer than it caches, it can be updated.